docs/operators/transform.md

```diff
@@ -0,0 +1,121 @@
+Transform - Operator Documentation
+==================================
+
+About Transform operator
+------------------------
+
+Transform means mapping of field expression from input to output or conversion of fields from one type to another.
+This operator is stateless. This operator can transform input POJO using provided expression mappings into output POJO.
+
+Use Case
+--------
+
+Consider the data received from input source and want to transform the data as below output schema while emitting to downstream operator. 
+
+Consider the below input schema
+`{FirstName, LastName, Phone, DateOfBirth, Address}`
+
+and output schema
+`{Name, Phone, Age, Address}`
+
+Suppose the requirement as follows:
+    -   "Name" field would be the combination of FirstName and LastName 
+    -   "Age" can be calculated from DateOfBirth.
+            
+Then, below are the expression mappings between output and input fields. 
+
+```
+Name => {$.FirstName}.concat(\" \").concat({$.LastName})
+Age => (new java.util.Date()).getYear() - {$.dateOfBirth}.getYear()
+```
+
+#### Configuration Parameters
+-----------------------------
+
+-   ***expressionMap*** -   Map<String, String>
+    -   Mandatory Parameter
+    -   Specifies the map between the output field and expression using fields of input POJO.
+
+-   ***expressionFunctions*** -   List<String>
+    -   List of imported classes or methods should be made available to expression to use.
+    -   Default Value = {java.lang.Math.*, org.apache.commons.lang3.StringUtils.*, org.apache.commons.lang3.StringEscapeUtils.*, org.apache.commons.lang3.time.DurationFormatUtils.*, org.apache.commons.lang3.time.DateFormatUtils.*}
+
+-   ***copyMatchingFields*** -   boolean
+    -   Specifies whether the matching fields between input and output should be copied as is or not.
+    -   Default Value = true.
+    
+#### Ports
+----------
+
+-   ***input*** -   Port for input tuples.
+-   ***output***    -   Port for transformed output tuples.
+
+#### Attributes
+---------------
+-   ***Input port Attribute - input.TUPLE\_CLASS*** - Fully qualified class name.
+    -   Mandatory attribute
+    -   Type of input tuple.
+
+-   ***Output port Attribute - output.TUPLE\_CLASS*** - Fully qualified class name.
+    -   Mandatory attribute
+    -   Type of output tuple.
+    
```

Comment:
Are there any restrictions on these input and output types, for instance, do they need to be Java serializable or Kryo serializable or both ?